test/e2e/framework/pv/pv.go

```diff
@@ -844,3 +848,21 @@ func WaitForPersistentVolumeDeleted(c clientset.Interface, pvName string, poll,
 	}
 	return fmt.Errorf("PersistentVolume %s still exists within %v", pvName, timeout)
 }
+
+// WaitForPVCFinalizer waits for a finalizer to be added to a PVC in a given namespace.
+func WaitForPVCFinalizer(ctx context.Context, cs clientset.Interface, name, namespace, finalizer string, poll, timeout time.Duration) error {
+	defer ginkgo.GinkgoRecover()
```

Comment:
Wondering why we're calling GinkgoRecover here? According to the docs:

```
//GinkgoRecover should be deferred at the top of any spawned goroutine that (may) call `Fail`
//Since Gomega assertions call fail, you should throw a `defer GinkgoRecover()` at the top of any goroutine that
//calls out to Gomega
//
//Here's why: Ginkgo's `Fail` method records the failure and then panics to prevent
//further assertions from running.  This panic must be recovered.  Ginkgo does this for you
//if the panic originates in a Ginkgo node (an It, BeforeEach, etc...)
//
//Unfortunately, if a panic originates on a goroutine *launched* from one of these nodes there's no
//way for Ginkgo to rescue the panic.  To do this, you must remember to `defer GinkgoRecover()` at the top of such a goroutine.
func GinkgoRecover() {
	e := recover()
	if e != nil {
		global.Failer.Panic(codelocation.New(1), e)
	}
}
```

Doesn't seem like this runs in a goroutine or calls Failf

Reply:
Good point! I had a `Failf` there in an earlier commit. Removed this and also the `ExpectNoError`'s from this function